deepmd/pt/model/model/ener_model.py

```diff
@@ -6,12 +6,23 @@
 
 import torch
 
+from deepmd.pt.model.atomic_model import (
+    DPAtomicModel,
+)
+from deepmd.pt.model.model.model import (
+    BaseModel,
+)
+
 from .dp_model import (
     DPModel,
 )
+from .make_model import (
+    make_model,
+)
 
 
-class EnergyModel(DPModel):
+@BaseModel.register("standard")
```

Comment:
Inconsistent with dipole

https://github.com/deepmodeling/deepmd-kit/blob/56accdfaa1779377d57913b1b05f77a011e69aee/deepmd/pt/model/model/dipole_model.py#L24-L25

Reply:
What is the proposed design here?

Reply:
the principle is that each model is derived from its own atomic model. we need to implement energy, dos, models at the atomic level. Then we can remove the ugly code in the dp_model
https://github.com/deepmodeling/deepmd-kit/blob/a26b680308f02b4c0a3648aac7b76bebef0ae0ef/deepmd/pt/model/model/dp_model.py#L48-L80

Reply:
I see what you mean. But is it good to move all stuff to the top module?

https://github.com/deepmodeling/deepmd-kit/blob/56accdfaa1779377d57913b1b05f77a011e69aee/deepmd/pt/model/model/__init__.py#L172-L181

Reply:
> Inconsistent with dipole
> 
> https://github.com/deepmodeling/deepmd-kit/blob/56accdfaa1779377d57913b1b05f77a011e69aee/deepmd/pt/model/model/dipole_model.py#L24-L25

registered as "energy" and "standard", in the numpy backend, we only have implementation for `EnergyModel` which is `standard`.

Reply:
>  In the numpy backend, we only have implementation for `EnergyModel` which is `standard`.

I think we don't even have `EnergyModel` in the numpy backend... Please check below.

https://github.com/deepmodeling/deepmd-kit/blob/a26b680308f02b4c0a3648aac7b76bebef0ae0ef/deepmd/dpmodel/model/dp_model.py#L20-L21

I believe it doesn't represent any specific property.

Reply:
I mean, is `standard` required by something?

Reply:
> I mean, is `standard` required by something?

I guess I will have to change all `standard` to `ener`, which also includes the default tag in argcheck?

Reply:
How does standard + dipole fitting work in the current implementation?

Reply:
> How does standard + dipole fitting work in the current implementation?

in pytorch backend, `standard` is basically an alias for `ener`. I don't think we can register a model as `standard` + `dipole`, the old standard model (which was the DPModel) is removed.

Reply:
So again, what's the reason for removing DPModel, especially `__new__`? I don't think `__new__` conflicts with `make_model` or other methods.

Reply:
> So again, what's the reason for removing DPModel, especially `__new__`? I don't think `__new__` conflicts with `make_model` or other methods.

In the old design, DPModel inherit from make_model(DPAtomicModel), we want more flexibility, such that each Model can inherit from it's corresponding make_model(AtomicModel).

Reply:
We can keep __new__ and plugin behaviors but just remove inheritance.

Reply:
> We can keep `__new__` and plugin behaviors but just remove inheritance.

In that case, how do we handle the `deserialize` method in DPModel? If we remove inheritance, there is no deserialize for base model to call. 

```
@classmethod
    def deserialize(cls, data: dict) -> "BaseBaseModel":
        """Deserialize the model.
    
        Parameters
        ----------
        data : dict
            The serialized data
    
        Returns
        -------
        BaseModel
            The deserialized model
        """
        if inspect.isabstract(cls):
>           return cls.get_class_by_type(data["type"]).deserialize(data)
```

Reply:
We should choose one of below:
(1) Deprecate and remove `standard` in serialization.
(2) Keep `standard`, but have an abstract class to redirect to the actual class.

We shouldn't mix them.